crates/wallet/src/errors/mod.rs

```diff
@@ -1,31 +1,61 @@
 use thiserror::Error;
-use tonic::transport;
-use nuts::error::Error as NutsError;
-
-
 
 #[derive(Error, Debug)]
 pub enum WalletError {
     #[error("Database error: {0}")]
     Database(#[from] rusqlite::Error),
-
     #[error("Transport error: {0}")]
     Transport(#[from] tonic::transport::Error),
-    
-    #[error("gRPC error: {0}")]
-    Grpc(#[from] Status),
-
     #[error("Amount overflow")]
     AmountOverflow,
-
     #[error("No matching keyset found")]
     NoMatchingKeyset,
-
     #[error("Proof not available")]
     ProofNotAvailable,
-
     #[error("Invalid public key: {0}")]
     InvalidPublicKey(String),
+    #[error("Invalid keyset ID: {0}")]
+    InvalidKeysetId(String),
+    #[error("gRPC error: {0}")]
+    Grpc(String),
+    #[error("Protocol error: {0}")]
+    Protocol(String),
+    #[error("Nut01 error: {0}")]
+    Nut01(String),
+    #[error("Nut02 error: {0}")]
+    Nut02(String),
+    #[error("DHKE error: {0}")]
+    Dhke(String),
```

Comment:
Don't use `String` as the internal repr of the error. Use the original type

Reply:
and you can use the thiserror `#[from]` macro for those that are only present one. Instead of manually writing the implementation